src/main/java/seedu/address/ui/MonthlyFeeListPanel.java

```diff
@@ -0,0 +1,125 @@
+
+    @FXML
+    private ListView<MonthlyFee> monthlyFeeListView;
+
+    /**
+     * Creates a {@code MonthlyFeeListPanel} with the given {@code ObservableList}.
+     */
+    public MonthlyFeeListPanel(ObservableList<Student> studentList) {
+        super(FXML);
+        monthlyFeeList = FXCollections.observableArrayList();
+
+        populateMonthlyFeeList(studentList);
+        setStudentListListener(studentList);
+        populateMonthlyFeeListView();
+    }
+
+    /**
+     * Set the student list to list for changes and update the monthly fee list accordingly.
+     * @param studentList Student list to listen for change.
+     */
+    private void setStudentListListener(ObservableList<Student> studentList) {
+        studentList.addListener((ListChangeListener<Student>) change -> {
+            while (change.next()) {
+                monthlyFeeList.clear();
+                populateMonthlyFeeList(studentList);
+                populateMonthlyFeeListView();
+            }
+        });
+    }
+
+    /**
+     * Populates the MonthlyFeeListViewCell with the given {@code ObservableList}.
+     */
+    private void populateMonthlyFeeListView() {
+        monthlyFeeListView.setItems(monthlyFeeList);
+        monthlyFeeListView.setCellFactory(listView -> new MonthlyFeeListViewCell());
+    }
+
+    /**
+     * Populates the monthlyFeeList with {@code MonthlyFee} for the current month + previous 2 months.
+     * @param studentList Student list to listen for change.
+     */
+    private void populateMonthlyFeeList(ObservableList<Student> studentList) {
+        LocalDateTime currMonthYear;
+        LocalDateTime nextMonthYear;
+
+        LocalDateTime now = LocalDateTime.now();
+        Month month = new Month(now.getMonth().getValue());
+        Year year = new Year(now.getYear());
+        currMonthYear = DateUtil.convertToLocalDate(month, year);
+        nextMonthYear = currMonthYear.plusMonths(1);
+
+        // Get fees for the current month + previous 2 months.
+        for (int i = 0; i < 3; i++) {
+            // Get current month value and add the result to the resulting string
+            double currMonthlyFee = getCurrMonthFee(studentList, currMonthYear, nextMonthYear);
+            monthlyFeeList.add(new MonthlyFee(currMonthlyFee, new Month(currMonthYear.getMonth().getValue()),
+                new Year(currMonthYear.getYear())));
+            // Update to previous month
+            currMonthYear = currMonthYear.minusMonths(1);
+            nextMonthYear = currMonthYear.plusMonths(1);
+        }
+    }
+
+    /**
+     * Gets the current month fee based on the start period and end period.
+     *
+     * @param studentList List of students.
+     * @param startPeriod Start Period of the month.
+     * @param endPeriod End Period of the month.
+     * @return Monthly fee for the current month.
+     */
+    private double getCurrMonthFee(ObservableList<Student> studentList, LocalDateTime startPeriod,
```

Comment:
maybe `getMonthFeeByPeriod`? `getCurrMonthFee` seems to imply the current month, which is today's month till next month, while this function takes it start period and endperiod. just for clarity!

Reply:
Yeap! Sounds better!